saleor/core/tasks.py

```diff
@@ -1,16 +1,55 @@
 import logging
 
+from celery import Task
 from celery.utils.log import get_task_logger
 from django.conf import settings
 from django.core.files.storage import default_storage
+from django.db import connections
 from django.db.models import Exists, OuterRef
 from django.utils import timezone
 
 from ..celeryconf import app
+from ..core.db.connection import allow_writer
 from .models import EventDelivery, EventPayload
 
 task_logger: logging.Logger = get_task_logger(__name__)
 
+
+class RestrictWriterDBTask(Task):
```

Comment:
You also need to prepare proper PR for multitenant. 
We also use custom tasks there. 
https://github.com/saleor/saleor-multitenant/blob/main/tenants/tasks.py#L7

Reply:
MT will not be using this class, RestrictWriterDBTask will be only used in Core repo to raise errors in tests when working on local changes, or log warnings in Celery when running tasks with unallowed writer queries. Patch to adjust MT is here: https://github.com/saleor/saleor-multitenant/pull/1529